MoreLinq/AtMost.cs

```diff
@@ -0,0 +1,64 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2016 Jonas Nyrup. All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    static partial class MoreEnumerable
+    {
+        /// <summary>
+        /// Returns true when the number of elements in the given sequence is smaller than
+        /// or equal to the given integer.
+        /// This method throws an exception if the given integer is negative.
+        /// </summary>
+        /// <remarks>
+        /// The number of items streamed will be less than or equal to the given integer
+        /// plus one.
+        /// </remarks>
+        /// <typeparam name="TSource">Element type of sequence</typeparam>
+        /// <param name="source">The source sequence</param>
+        /// <param name="count">The maximum number of items a sequence must have for this
+        /// function to return true</param>
+        /// <exception cref="ArgumentNullException">source is null</exception>
+        /// <exception cref="ArgumentOutOfRangeException">count is negative</exception>
+        /// <returns><c>true</c> if the number of elements in the sequence is smaller than
+        /// or equal to the given integer or <c>false</c> otherwise.</returns>
+        /// <example>
+        /// <code>
+        /// var numbers = { 123, 456, 789 };
+        /// var result = numbers.AtMost(4);
+        /// </code>
+        /// The <c>result</c> variable will contain <c>true</c>.
+        /// </example>
+        public static bool AtMost<TSource>(this IEnumerable<TSource> source, int count)
+        {
+            if (source == null) throw new ArgumentNullException("source");
+            if (count < 0) throw new ArgumentOutOfRangeException("count", "The count must not be negative.");
+
+            var collection = source as ICollection<TSource>;
+            if (collection != null)
+            {
+                return collection.Count <= count;
+            }
+
+            return source.Take(count + 1).Count() <= count;
```

Comment:
To know that `source` has *no more* than `count` elements.
If e.g. `source` has 2 elements, and `count = 1` `source.Take(count).Count() <= count` yields `true` which is not the answer we want, as it has more than 1 element.